### DISCUSSION TITLE
Ruff linter for all frameworks

### DISCUSSION THREAD

**Discussion by ArtemIsmagilov** [astral-sh/ruff#12512] [Q&A] [answered]
Do I understand correctly that in the future you plan to create a linter and formatter for all python frameworks? From the description I see that you support numpy, fastapi and so on. Maybe I didn't understand correctly. For example, i prefer use ruff 
instead of isort + black + flake8. However, I'm wondering how you plan to implement support for frameworks? And which ones exactly? There are a million of them. And is this correct, considering that, for example, we want to work with the python code style and not with all frameworks? In general, what are the goals of the stable version of ruff? is this a linter for all python frameworks - flask, Django, fastapi, numpy? or a linter taking into account all the code style utilities - black, flake8, isort. or how?

**Comment by MichaReiser** [comment, marked as answer]
> Do I understand correctly that in the future you plan to create a linter and formatter for all python frameworks? 

Luckily, most lint rules aren't framework-specific so that Ruff can be used with any framework. Depending on the framework, it may make sense to enable or disable specific rules because they only make sense in this context or because they don't play nicely with a specific set of packages. 

> From the description I see that you support numpy, fastapi and so on

Ruff comes with built-in rules for some widely used packages. Supporting custom rules for less used packages is probably something we want to do by supporting plugins https://github.com/astral-sh/ruff/issues/283

> we want to work with the python code style and not with all frameworks? 

I'm not sure if I understand that part, but we prefer a framework-agnostic style guide over maintaining many different style guides per framework (which style guide should be used if you use two frameworks together that have a conflicting style guide?)

**Reply by ArtemIsmagilov** [reply]
Thanks for the answer.
